controller/workitem.go

```diff
@@ -561,6 +564,227 @@ func getVersion(version interface{}) (int, error) {
 	return -1, nil
 }
 
+// ConvertWorkItemsToCSV is responsible for converting given []WorkItem model object into a
+// []string object containing a set of CSV formatted data lines and a header line with labels.
+// This methods combines all CSV data of all WITs into a single CSV
+func ConvertWorkItemsToCSV(ctx context.Context, app application.Application, wits []workitem.WorkItemType, wis []workitem.WorkItem) (string, error) {
+	// uuidStringCache stores the UUID cache for ID resolvings of the CSV conversion
+	uuidStringCache := map[string]string{}
+	if len(wits) != len(wis) {
+		return "", errs.Errorf("length mismatch of work items (%d) and work item types (%d)", len(wis), len(wits))
+	}
+	// csvGrid holds the final CSV in non-serialized form.
+	csvGrid := [][]string{}
+	// create the global column mapping by iterating over all fields of all WITs
+	// we're using seperate []string types as maps are not guaranteed to iterate
+	// consistently in the same order; we need to iterate multiple times over
+	// the columns with a stable order
+	columnKeys := []string{}
+	columnLabels := []string{}
+	alreadyProcessedWITs := id.Map{}
+	for i := 0; i < len(wits); i++ {
+		// we only process each WIT once
+		if _, ok := alreadyProcessedWITs[wits[i].ID]; ok {
+			continue
+		}
+		alreadyProcessedWITs[wits[i].ID] = struct{}{}
+		// retrieve fields for all WITs
+		fieldLabels, fieldKeys, err := extractWorkItemTypeFields(wits[i])
+		if err != nil {
+			return "", errs.Wrapf(err, "failed to retrieve fields for work item type: %s", wits[i].ID)
+		}
+		for j := 0; j < len(fieldLabels); j++ {
```

Comment:
I don't see where `j` is used anywhere except for indexing, please change to regular range loop.
```suggestion
		for fieldLabel := range fieldLabels {
```

Reply:
Fixed b401deb. See above.